dali/util/nvml.h

```diff
@@ -241,6 +241,43 @@ inline void Shutdown() {
   CUDA_CALL(nvmlShutdown());
 }
 
+
+class NvmlInstance {
+ public:
+  static NvmlInstance CreateNvmlInstance() {
+    return NvmlInstance(true);
+  }
+
+  explicit NvmlInstance(bool init = false) {
+    if (init) {
+      Init();
+      is_created_ = true;
+    }
+  }
+
+  NvmlInstance(const NvmlInstance &) = delete;
+
+  NvmlInstance &operator=(const NvmlInstance &) = delete;
+
+  inline NvmlInstance(NvmlInstance &&other) : is_created_(other.is_created_) {
+    other.is_created_ = false;
+  }
+
+  inline NvmlInstance &operator=(NvmlInstance &&other) {
+    std::swap(is_created_, other.is_created_);
+    other.~NvmlInstance();
```

Comment:
In that case, I think the destructor should set the flag to false, otherwise it may Shutdown twice, with the expicit call and when the moved instance goes out of scope. Omitting the explict call to the destructor should be fine, right?

Reply:
Ture that. Fixed.